Admin: expose the subscription orders

blocknotes opened the pull request:
Add a new tab in the member view of subscriptions to list the orders
processed by the current subscription.

Closes #129 

### Screenshot
![orders](https://user-images.githubusercontent.com/6893256/107005846-98b6cb00-6790-11eb-92f1-c1da150e2b5b.png)

aldesantis commented on app/views/spree/admin/subscription_orders/index.html.erb:
```diff
@@ -0,0 +1,96 @@
+          <tr class="admin_subscription_order">
+            <td><%= l order.created_at.to_date %></td>
+            <td><%= link_to order.number, edit_admin_order_path(order) %></td>
+            <td>
+              <% if order.user %>
+                <%= link_to order.email, edit_admin_user_path(order.user) %>
+              <% else %>
+                <%= order.email %>
+              <% end %>
+              <%= link_to_with_icon('email', t('spree.actions.send_email'), "mailto:#{order.email}", no_text: true) %>
+            </td>
```
Given that the user is always going to be the same, perhaps we can remove this column?

aldesantis replied:
Instead of the user, it would be useful to display the order's state if possible.

blocknotes replied:
Sure: https://github.com/solidusio-contrib/solidus_subscriptions/pull/192/commits/5cc4b5d32f50c591598ece6dbd81a2455e28ada2

aldesantis commented on config/routes.rb:
```diff
@@ -24,6 +24,7 @@
       post :skip, on: :member
       resources :installments, only: [:index, :show]
       resources :subscription_events, only: :index
+      resources :subscription_orders, only: :index
```
Can we use `/orders` as the path, for consistency with the path used for a user's order history in the default Solidus backend?

blocknotes replied:
Good point: https://github.com/solidusio-contrib/solidus_subscriptions/pull/192/commits/f8f1408e134acbe3860247658bf1949281a4a645

blocknotes commented:
I'm going on rebasing this PR to see if specs become green 😄

aldesantis reviewed (approved):
Thanks @blocknotes! In the interest of pushing this forward, I've rebased and removed the controller spec (we should move away from using those, and the feature spec already covers everything we need).

blocknotes commented:
> Thanks @blocknotes! In the interest of pushing this forward, I've rebased and removed the controller spec (we should move away from using those, and the feature spec already covers everything we need).

Good for me about removing that spec 👍 I added it for continuity with the existing tests.
Going to merge.
